soc/silabs/silabs_siwx91x/siwg917/siwx91x_nwp.c

```diff
@@ -35,45 +35,47 @@ BUILD_ASSERT(DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(195) ||
 	     DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(255) ||
 	     DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(319));
 
-extern const sli_si91x_set_region_ap_request_t default_US_region_2_4GHZ_configurations;
-extern const sli_si91x_set_region_ap_request_t default_EU_region_2_4GHZ_configurations;
-extern const sli_si91x_set_region_ap_request_t default_JP_region_2_4GHZ_configurations;
-extern const sli_si91x_set_region_ap_request_t default_KR_region_2_4GHZ_configurations;
-extern const sli_si91x_set_region_ap_request_t default_SG_region_2_4GHZ_configurations;
-extern const sli_si91x_set_region_ap_request_t default_CN_region_2_4GHZ_configurations;
+struct siwx91x_nwp_data {
+	char current_country_code[WIFI_COUNTRY_CODE_LEN];
+};
+
+struct siwx91x_nwp_config {
+	void (*config_irq)(const struct device *dev);
+	uint32_t stack_size;
+	uint8_t power_profile;
+};
 
-static char current_country_code[WIFI_COUNTRY_CODE_LEN];
 typedef struct {
 	const char *const *codes;
 	size_t num_codes;
 	sl_wifi_region_code_t region_code;
 	const sli_si91x_set_region_ap_request_t *sdk_reg;
 } region_map_t;
 
+extern const sli_si91x_set_region_ap_request_t default_US_region_2_4GHZ_configurations;
+extern const sli_si91x_set_region_ap_request_t default_EU_region_2_4GHZ_configurations;
+extern const sli_si91x_set_region_ap_request_t default_JP_region_2_4GHZ_configurations;
+extern const sli_si91x_set_region_ap_request_t default_KR_region_2_4GHZ_configurations;
+extern const sli_si91x_set_region_ap_request_t default_SG_region_2_4GHZ_configurations;
+extern const sli_si91x_set_region_ap_request_t default_CN_region_2_4GHZ_configurations;
+
 static const char *const us_codes[] = {
-	"AE", "AR", "AS", "BB", "BM", "BR", "BS", "CA", "CO", "CR", "CU", "CX",
-	"DM", "DO", "EC", "FM", "GD", "GY", "GU", "HN", "HT", "JM", "KY", "LB",
-	"LK", "MH", "MN", "MP", "MO", "MY", "NI", "PA", "PE", "PG", "PH", "PK",
-	"PR", "PW", "PY", "SG", "MX", "SV", "TC", "TH", "TT", "US", "UY", "VE",
-	"VI", "VN", "VU", "00"
-	/* Map "00" (world domain) to US region,
-	 * as using the world domain is not recommended
-	 */
```

Comment:
hmm... maybe we should keep this comment.:

        "SG", "MX", "SV", "TC", "TH", "TT", "US", "UY", "VE", "VI", "VN", "VU",
        /* Map World Domain ("00") to US region */
        "00"